### ISSUE TITLE
Refactor entity subrecord

### ISSUE THREAD

**Issue by TheSlimvReal** [Aam-Digital/ndb-core#1589]
Generally, I am not very happy with the current design. In my feeling there are ways to configure the entity subrecord component that don't make much sense (at least for now):

When listening to the `rowClick` event emitter, it is possible to still make the component navigate to another screen at the same time. At the moment we only listen to this event emitter in combination with clickMode="none" and I don't see a use case where this would be different.
When setting clickMode to navigate, then this would also happen when clicking the plus button. This does not make sense as the entity created with the newRecordFactory has not been saved yet and therefore can't be navigated to.
We currently only have the following scenarios:

The outside component handles displaying of existing and new entities -> the subrecord sould only forward these events
The subrecord should navigate to the clicked entity, a plus button is not visible
The subrecord handles all clicks using the popup form
Therefore, I think the current way of configuring the entity subrecord component is not very suitable as it requires more complexity than necessary and also can lead to errors when wrongly configured.

**Comment by sleidig** [comment]
Maybe we should resolve together with this the question of how we configure details view (in popup and navigated to): https://github.com/Aam-Digital/ndb-core/pull/2142#discussion_r1444627017

**Comment by TheSlimvReal** [comment]
Yes I think that would by itself resolve this because the outside component would not need to dictate what should happen. Maybe only the question whether it should open a popup or navigate.
